### PR TITLE
Widget Test: GreetingCard

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-65

### Description

 TEST CASES
- Valid Input
- Long User Name
- Empty Profile Picture URL

### Screenshots

greeting_card_test

### Tips

Any tips for testing it?

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [ ] All items in the author checklist is checked or not applicable
- [ ] All files including .lock and assets are reviewed
- [ ] Reasonable UI flows/cases related to the change are manually tested
- [ ] No commented-out code or unnecessary debug statements.
- [ ] No spelling or grammatical errors in code comments or documentation.
- [ ] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [ ] The branch is up-to-date with the target branch (e.g., staging).
- [ ] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: test/widgets/cards/greeting_card_test.dart [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+              profilePictureUrl: testProfilePictureUrl,
+            ),
+          );
+        },
+      );
+
+      // Assert
+      expect(find.text(testGreetingMessage), findsOneWidget);
+      expect(find.text(testUserName), findsOneWidget);
+      expect(find.byType(CircleAvatar), findsOneWidget);
```
Comment (quiteachaos) [inline test/widgets/cards/greeting_card_test.dart]: also 
expect(avatarWidget.backgroundImage, isNotNull);

File: test/widgets/cards/greeting_card_test.dart [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+              userName: testUserName,
+              profilePictureUrl: testProfilePictureUrl,
+            ),
+          );
+        },
+      );
+
+      // Assert
+      expect(find.text(testGreetingMessage), findsOneWidget);
+      expect(find.text(testUserName), findsOneWidget);
```
Comment (quiteachaos) [inline test/widgets/cards/greeting_card_test.dart]: only first name should be visible

File: test/widgets/cards/greeting_card_test.dart [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      expect(find.byType(CircleAvatar), findsOneWidget);
+    });
+
+    testWidgets(
```
Comment (quiteachaos) [inline test/widgets/cards/greeting_card_test.dart]: remove this test case as only first name should be visible.
Maybe you can try giving a very long first name

File: lib/widgets/cards/greeting_card.dart [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Text(
+                greetingMessage,
+                style: theme.textStyle.bodyRegular.copyWith(
+                  color: theme.colors.contrastMedium,
+                ),
+              ),
+              Text(
+                userName,
```
Comment (quiteachaos) [inline lib/widgets/cards/greeting_card.dart]: only first name should be visible
basically until the code finds the first space(' ')

**Review by quiteachaos — CHANGES_REQUESTED** [review]

File: test/widgets/cards/greeting_card_test.dart (hunk 1 above)
Comment (krushnakantanayak) [reply test/widgets/cards/greeting_card_test.dart]: done

File: lib/widgets/cards/greeting_card.dart (hunk 4 above)
Comment (krushnakantanayak) [reply lib/widgets/cards/greeting_card.dart]: done

File: test/widgets/cards/greeting_card_test.dart (hunk 3 above)
Comment (krushnakantanayak) [reply test/widgets/cards/greeting_card_test.dart]: done

File: test/widgets/cards/greeting_card_test.dart (hunk 2 above)
Comment (krushnakantanayak) [reply test/widgets/cards/greeting_card_test.dart]: done

**Review by quiteachaos — APPROVED** [review]

**Pull request merged**